src/modeler/loadFiles/readScenarioGroup.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2007-2025, RTE (https://www.rte-france.com)
+// See AUTHORS.txt
+// SPDX-License-Identifier: MPL-2.0
+// This file is part of Antares-Simulator,
+// Adequacy and Performance assessment for interconnected energy networks.
+//
+// Antares_Simulator is free software: you can redistribute it and/or modify
+// it under the terms of the Mozilla Public Licence 2.0 as published by
+// the Mozilla Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Antares_Simulator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// Mozilla Public Licence 2.0 for more details.
+//
+// You should have received a copy of the Mozilla Public Licence 2.0
+// along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+
+#include <filesystem>
+#include <fmt/format.h>
+#include <fstream>
+#include <string>
+
+#include <antares/logs/logs.h>
+#include <antares/solver/modeler/loadFiles/loadFiles.h>
+#include "antares/exception/RuntimeError.hpp"
+#include "antares/optimisation/linear-problem-data-impl/Scenario.h"
+#include "antares/scenarioGroupParser/ScenarioGroupParser.h"
+
+namespace Antares::Solver::LoadFiles
+{
+namespace
+{
+const std::filesystem::path SCENARIO_BUILDER_PATH = "input/data-series/modeler-scenariobuilder.dat";
+}
+
+Optimisation::ScenarioGroupRepository parseScenarioGroupRepository(std::ifstream file)
+{
+    Optimisation::ScenarioGroupRepository scenarioGroupRepository;
+    ScenarioGroupParser parser;
+    std::map<std::string, std::unique_ptr<Optimisation::LinearProblemDataImpl::Scenario>>
+      alreadyCreatedScenarios;
+    std::string line;
+    while (std::getline(file, line))
+    {
+        if (line.empty() || line[0] == '#')
+        {
+            continue; // Skip empty lines and comments
+        }
+        try
+        {
+            auto parsedLine = parser.parseLine(line);
+            if (!alreadyCreatedScenarios.contains(parsedLine.groupName))
+            {
+                alreadyCreatedScenarios[parsedLine.groupName] = std::make_unique<
+                  Optimisation::LinearProblemDataImpl::Scenario>(parsedLine.groupName);
+            }
```

Comment:
According to `std::map::insert`'s documentation,
> Inserts element(s) into the container, **if the container doesn't already contain an element with an equivalent key.** 

So there is no need to check whether the key is already present `if (!alreadyCreatedScenarios.contains(parsedLine.groupName))` if you used `insert` or even better [`emplace`](https://en.cppreference.com/w/cpp/container/map/emplace.html).